engine/core/src/main/java/org/datacleaner/job/builder/AbstractComponentBuilder.java

```diff
@@ -322,11 +324,50 @@ public B setConfiguredProperty(String configuredName, Object value) {
     public B setConfiguredProperty(ConfiguredPropertyDescriptor configuredProperty, Object value) {
         final boolean changed = setConfiguredPropertyIfChanged(configuredProperty, value);
         if (changed) {
+            if (configuredProperty.isInputColumn()) {
+                for (TransformedInputColumn<?> transformedInputColumn : getTransformedInputColumns(value)) {
+                    // Register change listener on all transformers providing values used for the input
+                    // column.
+                    getAnalysisJobBuilder().getTransformerComponentBuilders().stream().filter(
+                            transformer -> (isProvidingColumn(transformedInputColumn, transformer))).forEach(
+                                    transformer -> transformer.addChangeListener(
+                                            new ComponentBuilderTransformerChangeListener(this, configuredProperty)));
+                }
+            }
+
             onConfigurationChanged();
         }
         return (B) this;
     }
 
+    protected boolean isProvidingColumn(final TransformedInputColumn<?> transformedInputColumn,
```

Comment:
Not sure if it makes sense, but if this was a method on TransformedInputColumn, you could use a method reference above.